client/db/bolt/upgrades_test.go

```diff
@@ -25,6 +25,7 @@ var dbUpgradeTests = [...]struct {
 	filename   string // in testdata directory
 	newVersion uint32
 }{
+	// {"testnetbot", v1Upgrade, verifyV1Upgrade, "dexbot-testnet.db.gz", 3}, // only for TestUpgradeDB
 	{"upgradeFromV0", v1Upgrade, verifyV1Upgrade, "v0.db.gz", 1},
 	{"upgradeFromV1", v2Upgrade, verifyV2Upgrade, "v1.db.gz", 2},
 	{"upgradeFromV2", v3Upgrade, verifyV3Upgrade, "v2.db.gz", 3},
```

Comment:
Test a big upgrade by commenting out these three and uncommenting the other one, which is only for `TestUpgradeDB`.

Then see how it goes with `go test -v -count 1 -run TestUpgradeDB`.

If it succeeds, you can try a mem profile:

```
go test -v -count 1 -run TestUpgradeDB -memprofile mem.out
go tool pprof bolt.test mem.out
# top and web are easiest
```

Reply:
does not oom for me

Reply:
Thanks for checking.  I suspect breaking each upgrade into different db txns allowed cleanup (GC or other OS business).  So I guess this PR is needed then.